scripts/create-kubestellar-demo-env.sh

```diff
@@ -0,0 +1,149 @@
+output=$(./check_pre_req.sh -V kflex ocm helm kubectl docker kind)
+
+if echo "$output" | grep -q "X"; then
+    echo "Please install all the necessary dependencies before contiuning"
+    exit 1
+else
+    echo "Pre-req are properly installed. Proceeding to environment clean up"
+fi
+
+echo -e "\nStarting environment clean up..."
+echo -e "Starting cluster clean up..."
+
+cluster_clean_up() {
+    error_message=$(eval "$1" 2>&1)
+    if [ $? -ne 0 ]; then
+        echo "clean up failed. Error:"
+        echo "$error_message"
+    fi
+}
+
+cluster_clean_up "kind delete cluster --name kubeflex"
+cluster_clean_up "kind delete cluster --name cluster1"
+cluster_clean_up "kind delete cluster --name cluster2"
+echo -e "Cluster space clean up has been completed"
+
+echo -e "\nStarting context clean up..."
+
+context_clean_up() {
+    output=$(kubectx)
+    # echo $output
+    while IFS= read -r line; do
+        if [ "$line" == "kind-kubeflex" ]; then 
+            echo "Deleting kind-kubeflex context..."
+            kubectl config delete-context kind-kubeflex
+
+        elif [ "$line" == "cluster1" ]; then
+            echo "Deleting cluster1 context..."
+            kubectl config delete-context cluster1
+
+        elif [ "$line" == "cluster2" ]; then
+            echo "Deleting cluster2 context..."
+            kubectl config delete-context cluster2
+        
+        elif [ "$line" == "its1" ]; then
+            echo "Deleting its1 context..."
+            kubectl config delete-context its1
+        
+        elif [ "$line" == "wds1" ]; then
+            echo "Deleting wds1 context..."
+            kubectl config delete-context wds1
+        
+        fi
+
+    done <<< "$output"
+}
+
+context_clean_up
+echo "Context space clean up completed"
+
+echo -e "\nStarting the process to install kubestellar core: kind-kubeflex..."
+export KUBESTELLAR_VERSION=0.23.1
+
+curl -s -o create-kind-cluster-with-SSL-passthrough.sh https://raw.githubusercontent.com/kubestellar/kubestellar/v${KUBESTELLAR_VERSION}/scripts/create-kind-cluster-with-SSL-passthrough.sh
+
+chmod +x create-kind-cluster-with-SSL-passthrough.sh
+
+./create-kind-cluster-with-SSL-passthrough.sh --name kubeflex --port 9443
+
+helm upgrade --install ks-core oci://ghcr.io/kubestellar/kubestellar/core-chart --version $KUBESTELLAR_VERSION --set-json='ITSes=[{"name":"its1"}]' --set-json='WDSes=[{"name":"wds1"}]'
+
+echo -e "\nWaiting for new non-host KubeFlex Control Planes to be Ready:"
+for cpname in its1 wds1; do
+  while [[ `kubectl get cp $cpname -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}'` != "True" ]]; do
+    echo "Waiting for \"$cpname\"..."
+    sleep 5
+  done
+  echo "\"$cpname\" is ready."
+done
+
+kubectl config delete-context its1 || true
+kflex ctx its1
+kubectl config delete-context wds1 || true
+kflex ctx wds1
+kflex ctx
+
+echo -e "\nCreating cluster and context for cluster 1 and 2..."
+: set flags to "" if you have installed KubeStellar on an OpenShift cluster
```

Comment:
That's not actually something that an invoker of this script can do.

Reply:
Perhaps this script should have a command-line flag that controls this behavior. Or auto-detect whether the cluster is OpenShift.

Reply:
Im will comment/delete out this line at the moment because I am not sure how openshift clusters work.

Reply:
I think that leaving a comment will be more helpful than just deleting the line.

Reply:
added